lib/core/parser/driver.cc

```diff
@@ -21,6 +21,17 @@ void Driver::parse(std::istream &iss) {
   parse_helper(iss);
 }
 
+uint64_t Driver::GetInterning(std::string &input_string) {
+  return interning_.GetInterning(input_string);
+}
+
+uint64_t Driver::Interning::GetInterning(std::string &input_string) {
+  if (!tokens_.contains(input_string)) {
+    tokens_[input_string] = counter_++;
+  }
+  return tokens_[input_string];
+}
+
```

Comment:
I have also considered this idea. I think that other components may want to get the token of the string without losing control over it. Is this situation possible?

Reply:
Also I think that this way we choose optimization over readability. Is the optimization here is really that needed?

Reply:
> Also I think that this way we choose optimization over readability. Is the optimization here is really that needed?

I mean we use interning to avoid using a costly string comparison and instead want to compare integers. As I see it we transform the strings into integers and use them, so I can't come up with a situation where we would need to continue holding onto the string. If we do come across such a case it would be a matter of making another overload, shouldn't be too cumbersome.